### PR TITLE
Feature/sbom processor

### PR DESCRIPTION
**Pull request by dzakirin-muhammad**
merging sbom-processor into main

### CODE REVIEW

File: structs/Filesystem.go [hunk 1]
```diff
@@ -48,10 +49,10 @@ func (filesystem *Filesystem) Scan() []ResultInfo {
 
 	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
```
Comment (hazimhasnan-edag) [inline structs/Filesystem.go:50]: what happened to using WalkDir? @dzakirin-muhammad

Comment (dzakirin-muhammad) [reply structs/Filesystem.go:50]: will commit shortly

**Review by hazimhasnan-edag — APPROVED** [review]

File: adapters/maven/Schema.go [hunk 2]
```diff
@@ -10,10 +10,12 @@ type Build struct {
 }
 
 type Configuration struct {
-	XMLName xml.Name `xml:"configuration,omitempty"`
-	Url     string   `xml:"url,omitempty"`
-	Timeout int      `xml:"timeout,omitempty"`
-	Options []string `xml:"options>option,omitempty"`
+	Value string `xml:",innerxml"`
+	//XMLName  xml.Name  `xml:"configuration,omitempty"`
```
Comment (soon-yoong) [inline adapters/maven/Schema.go:14]: i think we can remove these commented codes

Comment (dzakirin-muhammad) [reply adapters/maven/Schema.go]: Removed unused commented code

**Review by norazlina-md-nor — APPROVED** [review]

**Review by soon-yoong — APPROVED** [review]

**Pull request merged**